### PR TITLE
Actively split pick tasks as early as possible.

### PR DESCRIPTION
**Pull request by Tilps**
Should reduce picking latency.

NPS tests not started yet - but fixed node test underway is 'fine' so far.

### CODE REVIEW

**Comment by Tilps** [comment]
Full benchmark results are not super conclusive but maybe 1% faster.  Startposition go 20M nodes was definitely in the noise.

It seems the 'correct' thing to do, I suspect that it'll make more of a difference on a slower cpu machine. Doesn't seem any real downsides.

**Comment by Tilps** [comment]
Retested post merge - 1% improvement was maintained.

File: src/mcts/search.cc [hunk 1]
```diff
@@ -1804,6 +1775,42 @@ void SearchWorker::PickNodesToExtendTask(Node* node, int base_depth,
         }
       }
       is_root_node = false;
+      // Actively do any splits now rather than waiting for potentially long tree walk to get there.
```
Comment (mooskagh) [inline src/mcts/search.cc:1778]: Line too long. :-P

**Review by mooskagh — APPROVED** [review]

File: src/mcts/search.cc (hunk 1 above)
Comment (Tilps) [reply src/mcts/search.cc]: Fixed.

**Pull request merged**